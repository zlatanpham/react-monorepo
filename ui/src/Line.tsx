import React from 'react';
import styled from 'styled-components';
import tw from 'tailwind.macro';

const Container = styled.div`
  ${tw`hidden`};
  height: 200px;
  background-color: red;
`;

type LineThickness = 'thin' | 'base' | 'thick';
interface LineProps {
  thickness: LineThickness;
}

export const Line: React.FunctionComponent<LineProps> = ({ thickness }) => {
  const weight: { [k in LineThickness]: number } = {
    base: 300,
    thick: 900,
    thin: 500,
  };
  return <Container style={{ fontWeight: weight[thickness] }}>Line</Container>;
};
