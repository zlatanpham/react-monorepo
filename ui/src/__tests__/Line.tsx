import React from 'react';
import { render } from '@testing-library/react';
import { Line } from '../Line';
import 'jest-styled-components';

describe('<Line/>', () => {
  it('Should match snapshot', () => {
    const { container } = render(<Line thickness="thick" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
