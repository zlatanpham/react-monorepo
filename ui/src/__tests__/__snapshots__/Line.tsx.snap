// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Line/> Should match snapshot 1`] = `
<div
  class="sc-bdVaJa fhAnNi"
  style="font-weight: 900;"
>
  Line
</div>
`;
